src/config.rs

```diff
@@ -74,22 +78,27 @@ impl Default for Config {
             enable_linter: true,
             run_test_command: String::new(),
             test_runner: TestRunner::Auto,
+            was_reloaded: false,
         }
     }
 }
 
 impl Config {
     /// Reloads the configuration from the language client.
-    pub fn reload(
-        &mut self,
-        requester: &mut Requester<'_>,
-        client_capabilities: &ClientCapabilities,
-    ) -> LSPResult<()> {
-        if !client_capabilities.workspace_configuration_support() {
+    /// This method is static to trick the borrow checker - [`Config`] is an element of [`State`].
+    pub fn reload(state: &mut State, requester: &mut Requester<'_>) -> LSPResult<()> {
+        if !state.client_capabilities.workspace_configuration_support() {
             warn!(
                 "client does not support `workspace/configuration` requests, config will not be \
                  reloaded"
             );
+
+            if !state.config.was_reloaded {
+                Self::on_config_change(state)
+            }
+
+            state.config.was_reloaded = true;
```

Comment:
```suggestion
            let was_reloaded = state.config.was_reloaded;
            state.config.was_reloaded = true;
```

And move to the beginning of the function, this way you can have it done in a single place, making it easier to handle in case we add here more complex control flow